fixtures/tests/_fixtures/test_monkeypatch.py

```diff
@@ -72,14 +83,112 @@ def test_delete_missing_attribute(self):
             fixture.cleanUp()
             self.assertFalse('new_attr' in globals())
 
+    def _check_static_or_class_method(self, name, new_method):
```

Comment:
Correct. I ran into a number of interesting cases in Nova while trying to work out a good solution. Depending on the combination of static vs instance method used for patching and Python version I was finding different unexpected behaviors. For instance while in the scope of the fixture a method might be callable but then after the fixture cleanup it may not be. Or it was no longer callable on the class because it had lost the descriptor.

I didn't run across it while testing on Nova but classmethods would pose the same challenge.